specifications/wallet/spec-light-mode.md

```diff
@@ -0,0 +1,123 @@
+
+In addition, the "trust versus speed" trade-off will be partially obsoleted by [Mithril][] technology, which aims to give us "trust *and* speed" by providing verified ledger state data as opposed to block data. The act of implementing light-mode in `cardano-wallet` does not only offer immediate benefits, but will, in fact, partially prepare the codebase for Mithril technology.
+
+  [daedalus]: https://daedaluswallet.io
+  [nami]: https://namiwallet.io
+  [blockfrost]: http://blockfrost.io
+  [mithril]: https://iohk.io/en/research/library/papers/mithrilstake-based-threshold-multisignatures/
+
+# Motivation
+
+## Background
+
+As the Cardano blockchain grows in size, retrieving and verifying blocks from the consensus network becomes increasingly time consuming. By making a "trust versus speed" trade-off, we can significantly decrease synchronization times.
+
+## Benefits
+
+* *Speed*. With light-mode, we expect synchronisation times of < 1 minute for a wallet with 1'000 transactions. In contrast, synchronisation of an empty wallet currently takes ~ 50 minutes by connecting to a node.
+* *Compatiblity*. Light-mode is intended to preserve the API presented to downstream software such as [Daedalus][], only *minimal* changes are expected.
+* *Optionality*. A wallet that was started in light-mode can be restarted in full mode without resynchronization, and vice versa. (MVP: no support yet for changing the mode dynamically while the wallet is running.)
+
+## Limitations
+
+* *Trust*. The external data source does *not* provide the same level of protection against *omission* of transactions as the Proof-of-Stake consensus protocol.
+* *Privacy*. In addition to the reduction of trust in the blockchain data, we now also reveal private information about addresses belonging to a single wallet.
+* *Address schemes*. Only *Shelley* wallets with *sequential address discovery* can be supported.
+
+However, these limitations are shared by all existing light wallets. In fact, some light wallets only provide *single-address* wallets, which is an additional reduction of privacy.
+
+# Specification
+
+## Functionality
+
+### Command line
+
+The `cardano-wallet` executable will feature a new flag `--light` which indicates that the executable is to be started in light-mode:
+
+```
+$ cardano-wallet serve --light CRED
+```
+
+* `CRED` specifies how to connect to the less trusted blockchain data source. (MVP: Use [Blockfrost][]; `CRED` is a filepath to the secret token.)
+* The `--light` argument and the `--node-socket` arguments are mutally exclusive with each other.
+
+### REST API
+
+When the executable was started in light-mode:
+
+* The endpoints in the hierarchy `/v2/byron-wallets/*` MUST return an error.
```

Comment:
There should be an explanation somewhere of why this is.
"... Because byron wallets use random derivation indices, they are significantly more challenging to make a light wallet for."

However we do have options should a byron light wallet really be desired:
1. Develop a third-party byron address discovery service which takes a byron root XPub and sends back a list of addresses belonging to that wallet. (May not be a great idea security-wise).
2. For existing byron wallets in Daedalus, their wallet database contains addresses generated/discovered so far. Without the node running, we can use these addresses for light wallet transaction searching. So the light wallet balance would be up-to-date, as long as no new addresses have been generated/used since that wallet was last fully synced.